src/iap/shared/components/carousel/carousel.component.ts

```diff
@@ -0,0 +1,63 @@
+import { Component, Input, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { Slide, SlideComponent } from '@shared/models/carousel.models';
+import { SlideDirective } from '../../directives/slide.directive';
+
+@Component({
+	selector: 'iap-carousel',
+	templateUrl: './carousel.component.html',
+	styleUrls: ['./carousel.component.scss'],
+})
+export class CarouselComponent implements OnInit, OnDestroy {
+	@ViewChild(SlideDirective, { static: true }) slideHost!: SlideDirective;
+
+	@Input()
+	slides: Slide<any>[] = [];
+
+	currentSlideIndex = 0;
+
+	interval: NodeJS.Timeout | undefined;
+
+	ngOnInit(): void {
+		this.selectSlide(this.currentSlideIndex);
+		this.startSlideshow();
+	}
```

Comment:
Please, add to this place `// TODO: ngOnChanges` section. We'll discuss one interesting problem later. :smirk: